src/python/include/Persistent_cohomology_interface.h

```diff
@@ -73,6 +72,54 @@ persistent_cohomology::Persistent_cohomology<FilteredComplex, persistent_cohomol
     return persistence;
   }
 
+  int top_dimensional_coface(int splx){
+    if (stptr_->dimension(splx) == stptr_->dimension()){return splx;}
+    else{  
+      for (auto v : stptr_->coboundary_simplex_range(splx)){
+        if(stptr_->filtration(v) == stptr_->filtration(splx)){
+          return top_dimensional_coface(v);
+        }
+      }  
+    }
+  }
```

Comment:
To remove the warning, in case filtration is not found, you shall return something (splx ?):
```suggestion
  int top_dimensional_coface(int splx){
    if (stptr_->dimension(splx) == stptr_->dimension())
      return splx;
    
    for (auto v : stptr_->coboundary_simplex_range(splx)) {
      if(stptr_->filtration(v) == stptr_->filtration(splx)) {
        return top_dimensional_coface(v);
      }
    }
    return splx;
  }
```

Reply:
`BOOST_UNREACHABLE_RETURN(something)`?

Reply:
I followed Vincent's suggestion.

Reply:
I don't like it, we are generating more code, and if there is a bug and we reach it, we are returning a value that looks possible. If we are going to return a value, at least make it something more likely to explode (`return -1; // Unreachable`?).